ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -83,6 +83,14 @@ public UIManagerModule(
     reactContext.addLifecycleEventListener(this);
   }
 
+  /**
+   * This method gives an access to the underlying native view hierarchy manager. Note that all the
+   * operations executed on that object should be run in the UI thread. Use with caution
+   */
+  public NativeViewHierarchyManager getNativeViewHierarchyManager() {
+    return mUIImplementation.getUIViewOperationQueue().getNativeViewHierarchyManager();
```

Comment:
Ok, I feel convinced that this should be a separate module (if only because of Rebound). Exposing the NativeViewHierarchyManager seems like the only reasonable way to do this that I can think of unfortunately...